components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -802,6 +803,26 @@ protected void unsetClaimFilter(ClaimFilter claimFilter) {
         FrameworkServiceDataHolder.getInstance().removeClaimFilter(claimFilter);
     }
 
+    @Reference(
+            name = "approles.resolver.service",
+            service = ApplicationRolesResolver.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetAppRolesResolverService"
+    )
+    protected void setAppRolesResolverService(ApplicationRolesResolver applicationRolesResolver) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Application Roles Resolver is set in the Application Authentication Framework bundle.");
+        }
```

Comment:
move the debug log after line 818.
log.isDebugEnabled() is not required, because internally even log.debug() method check that condition.
If there is no string concatenation/ any other resource consuming operation need to be performed for debug log we can add this check

Reply:
Addressed in 734a7bff58df0ecbf63400a8d7f3f61cf3ff22b4